### PR TITLE
PLF-6709: Backport - Property exo:lastModifiedDate not found when the daily and weekly notifications jobs are executed

### PR DESCRIPTION
**Pull request by azayati**

### CODE REVIEW

File: commons-component-common/src/main/java/org/exoplatform/commons/notification/impl/service/storage/NotificationDataStorageImpl.java [hunk 1]
```diff
@@ -245,6 +246,11 @@ private NodeIterator getDailyNodes(Node pluginDayNode, String userId) throws Exc
 
   private NotificationInfo fillModel(Node node) throws Exception {
     if(node == null) return null;
+    if(!node.hasProperty(EXO_LAST_MODIFIED_DATE) && node.canAddMixin(EXO_MODIFY)){
+      node.addMixin(EXO_MODIFY);
+      node.setProperty(EXO_LAST_MODIFIED_DATE, Calendar.getInstance());
+      node.save();
+    }
```
Comment (boubaker) [inline commons-component-common/src/main/java/org/exoplatform/commons/notification/impl/service/storage/NotificationDataStorageImpl.java:262]: Use the same strategy here:

if(!node.hasProperty(EXO_LAST_MODIFIED_DATE)) {
if(node.isNodeType(EXO_MODIFY)) {
node.setProperty(EXO_LAST_MODIFIED_DATE, Calendar.getInstance());
node.save();
}
else if(node.canAddMixin(EXO_MODIFY)) {
node.addMixin(EXO_MODIFY);
node.setProperty(EXO_LAST_MODIFIED_DATE, Calendar.getInstance());
node.save();
}
else {
LOG.warn("Cannot add mixin to node '{}'.", node.getPath());
}
}

File: commons-component-common/src/main/java/org/exoplatform/commons/notification/impl/service/storage/NotificationDataStorageImpl.java [hunk 2]
```diff
@@ -254,7 +260,7 @@ private NotificationInfo fillModel(Node node) throws Exception {
       .setSendToWeekly(NotificationUtils.valuesToArray(node.getProperty(NTF_SEND_TO_WEEKLY).getValues()))
       .setLastModifiedDate(node.getProperty(EXO_LAST_MODIFIED_DATE).getDate())
       .setId(node.getName());
-
+
```
Comment (boubaker) [inline commons-component-common/src/main/java/org/exoplatform/commons/notification/impl/service/storage/NotificationDataStorageImpl.java]: Don't change this line

File: commons-component-common/src/main/java/org/exoplatform/commons/notification/impl/service/storage/NotificationDataStorageImpl.java (hunk 1 above)
Comment (azayati) [inline commons-component-common/src/main/java/org/exoplatform/commons/notification/impl/service/storage/NotificationDataStorageImpl.java:262]: done

File: commons-component-common/src/main/java/org/exoplatform/commons/notification/impl/service/storage/NotificationDataStorageImpl.java (hunk 2 above)
Comment (azayati) [inline commons-component-common/src/main/java/org/exoplatform/commons/notification/impl/service/storage/NotificationDataStorageImpl.java]: done

**Pull request closed without merging**